docs/verify.rst

```diff
@@ -83,13 +83,16 @@ desired interface.
    True
 
 If all instances will provide the interface, we can
-mark the class as implementing it.
+mark a class as implementing it.
 
 .. doctest::
 
+   >>> class Foo2(object):
+   ...     x = 1
+   ...     y = 2
    >>> from zope.interface import classImplements
-   >>> classImplements(Foo, IFoo)
-   >>> verify_foo()
```

Comment:
Interesting. So the error here on `classImplements(Foo, IFoo)` was 
```
InconsistentResolutionOrderError: 
    For object directlyProvides(Foo, IFoo, classImplements(Foo, IFoo)).
```

The underlying cause is the existence of the `foo` object that already had `alsoProvides(foo, IFoo)` called on it. When the declaration on the `Foo` *class* was changed, the declaration (`directlyProvides`) on the `foo` *instance* has to be updated because it depends on the class definition. The C3 algorithm can't find a consistent ordering for the instance.

In this case, another way to solve the problem is to simply discard the instance (`del foo`), or note that the instance no longer implements the interface directly (`noLongerProvides(foo, IFoo)`). In either case, because of the use of weakrefs, it's necessary to run `gc.collect()` once or twice after taking the action.

I like the `noLongerProvides(foo, IFoo)` approach, because we can then continue to use the same `verify_foo()` call on the same instance object, preserving the original intent of the example.

I wonder if there's something we can do in general about this. For example, when we're implicitly updating a `directlyProvides` instance, can we automatically remove interfaces already present in a base (or at least, the `classImplements` base)? This would be the moral equivalent of `noLongerProvides` being called.